### ISSUE TITLE
Apply changelog badges backwards

### ISSUE THREAD

**Issue by ymzayek** [nilearn/nilearn#3578]
In #3542 we started adding badges to the changelog. This was applied backwards up to v0.9.0, but we wish to apply them fully to all the changelog files in `doc/changes` with the exception of `latest.rst` which will be handled during a release. This issue can be a reference for this. This can be done across several PRs by several contributors if it's too much to handle all at once. So anybody can take part by commenting to say which file(s) they will work on and open a pull request referencing this issue.

Badge mapping:

    :bdg-primary:`Doc`
    :bdg-secondary:`Maint`
    :bdg-success:`API`
    :bdg-info:`Plotting`
    :bdg-warning:`Test`
    :bdg-danger:`Deprecation`
    :bdg-dark:`Code`

Unassigned badge if a new category is warranted:

    :bdg-light:``

### Affected pages

Dev docs with some badges already added: https://nilearn.github.io/dev/changes/whats_new.html

See PR #3542 for a code reference on how badges are applied

**Comment by ymzayek** [comment]
@Remi-Gau yes that sounds good. Looking at the tags already placed we could probably replace `` :bdg-dark:`Code` `` with one for bug fixes too

**Comment by Remi-Gau** [comment]
Actually most of our changelog already have a section "fixes" so having a label for bug fixes may be redundant.
But your suggestion makes me think that we could use the code label as some sort of miscellaneous label.

**Comment by ymzayek** [comment]
Yes that was the initial thought and I also had the same thought about redundancy. So it was kind of a placeholder until we can think of something more informative. But we can leave it as is, no strong opinion here

**Comment by emdupre** [comment]
It looks like [sklearn structures their changelog by module](https://scikit-learn.org/stable/whats_new/v1.3.html), which I suppose is one alternative to avoid redundancy ?
